test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Rendering/HtmlHelperDropDownListExtensionsTest.cs

```diff
@@ -0,0 +1,256 @@
+            {
+                new SelectListItem { Text = "Four", Value = "4" },
+                new SelectListItem { Text = "Five", Value = "5" },
+            };
+            var metadataProvider = new EmptyModelMetadataProvider();
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper(new ViewDataDictionary<TestModel>(metadataProvider));
+            helper.ViewContext.ClientValidationEnabled = false;
+            helper.ViewData.ModelState.SetModelValue("Property1", 4, "4");
+            helper.ViewData["Property1"] = BasicSelectList;
+
+            // Act
+            var dropDownListResult = helper.DropDownList("Property1", selectList, new { Key = "Value", name = "CustomName" });
+
+            // Assert
+            Assert.Equal(expectedHtml, HtmlContentUtilities.HtmlContentToString(dropDownListResult));
+        }
+
+        [Fact]
+        public void DropDownList_UsesExpressionAndSpecifiedSelectListAndOptionLabel()
+        {
+            // Arrange
+            var expectedHtml = "<select id=\"HtmlEncode[[Property2]]\" name=\"HtmlEncode[[Property2]]\">" +
+                "<option value=\"\">HtmlEncode[[--select--]]</option>" + Environment.NewLine +
+                "<option value=\"HtmlEncode[[4]]\">HtmlEncode[[Four]]</option>" + Environment.NewLine +
+                "<option selected=\"HtmlEncode[[selected]]\" value=\"HtmlEncode[[5]]\">HtmlEncode[[Five]]</option>" + Environment.NewLine +
+                "</select>";
+            var selectList = new List<SelectListItem>
+            {
+                new SelectListItem { Text = "Four", Value = "4" },
+                new SelectListItem { Text = "Five", Value = "5" },
+            };
+            var metadataProvider = new EmptyModelMetadataProvider();
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper(new ViewDataDictionary<TestModel>(metadataProvider));
+            helper.ViewContext.ClientValidationEnabled = false;
+            helper.ViewData.Model = new TestModel { Property2 = "5" };
+
+            // Act
+            var dropDownListResult = helper.DropDownList("Property2", selectList, "--select--");
+
+            // Assert
+            Assert.Equal(expectedHtml, HtmlContentUtilities.HtmlContentToString(dropDownListResult));
+        }
+
+        [Fact]
+        public void DropDownListFor_NullSelectListFindsListFromViewData()
+        {
+            // Arrange
+            var expectedHtml = "<select id=\"HtmlEncode[[Property1]]\" name=\"HtmlEncode[[Property1]]\">" +
+                "<option value=\"HtmlEncode[[0]]\">HtmlEncode[[Zero]]</option>" + Environment.NewLine +
+                "<option value=\"HtmlEncode[[1]]\">HtmlEncode[[One]]</option>" + Environment.NewLine +
+                "<option value=\"HtmlEncode[[2]]\">HtmlEncode[[Two]]</option>" + Environment.NewLine +
+                "<option value=\"HtmlEncode[[3]]\">HtmlEncode[[Three]]</option>" + Environment.NewLine +
+                "</select>";
```

Comment:
Good that an `<option>` has `selected="selected"` in every `DropDownList()` test. Use the `SetModelValue()` trick to do the same here.